provider/openstack/provider.go

```diff
@@ -2124,6 +2063,14 @@ func (e *Environ) terminateInstances(ctx context.ProviderCallContext, ids []inst
 				break
 			}
 		}
+
+		// Attempt to destroy the ports that could have been created when using
+		// spaces.
+		if err := e.terminateInstanceNetworkPorts(id); err != nil {
```

Comment:
If we destroy the server first, the ports will no longer be associated with it.
We need to either delete them before, or collect the port IDs before.